test/Microsoft.Sbom.Targets.Tests/GenerateSbomTaskTests.cs

```diff
@@ -5,37 +5,271 @@
+    private const string PackageSuplier = "Test-Microsoft";
+    private const string PackageName = "CoseSignTool";
+    private const string PackageVersion = "0.0.1";
+    private const string NamespaceBaseUri = "https://base0.uri";
+
+    private Mock<IBuildEngine> buildEngine;
+    private List<BuildErrorEventArgs> errors;
+    private ISBOMValidator sbomValidator;
+    private string manifestPath;
 
     [TestInitialize]
     public void Startup()
     {
+        var host = Host.CreateDefaultBuilder()
+            .ConfigureServices((host, services) =>
+                services
+                .AddSbomTool())
+            .Build();
+        this.sbomValidator = host.Services.GetRequiredService<ISBOMValidator>();
+
         // Setup the build engine
-        buildEngine = new Mock<IBuildEngine>();
-        errors = new List<BuildErrorEventArgs>();
-        buildEngine.Setup(x => x.LogErrorEvent(It.IsAny<BuildErrorEventArgs>())).Callback<BuildErrorEventArgs>(e => errors.Add(e));
+        this.buildEngine = new Mock<IBuildEngine>();
+        this.errors = new List<BuildErrorEventArgs>();
+        this.buildEngine.Setup(x => x.LogErrorEvent(It.IsAny<BuildErrorEventArgs>())).Callback<BuildErrorEventArgs>(e => errors.Add(e));
 
         // Clean up the manifest directory
         if (Directory.Exists(ManifestDirectory))
         {
             Directory.Delete(ManifestDirectory, true);
         }
+
+        this.manifestPath = Path.Combine(ManifestDirectory, "spdx_2.2", "manifest.spdx.json");
+    }
+
+    [TestCleanup]
+    public void Cleanup()
+    {
+#pragma warning disable VSTHRD002 // Avoid problematic synchronous waits
+        if (File.Exists(this.manifestPath))
+        {
+            Assert.IsTrue(this.sbomValidator.ValidateSbomAsync(
+                CurrentDirectory,
+                TemporaryDirectory,
+                null,
+                manifestDirPath: Path.Combine(this.manifestPath, "..", "..")).GetAwaiter().GetResult().IsSuccess);
+        }
+#pragma warning restore VSTHRD002 // Avoid problematic synchronous waits
+
+        // Clean up the manifest directory
+        if (Directory.Exists(TemporaryDirectory))
+        {
+            Directory.Delete(TemporaryDirectory, true);
+        }
     }
 
     [TestMethod]
     public void Sbom_Is_Successfully_Generated()
+    {
+        // Arrange
+        var task = new GenerateSbomTask
+        {
+            BuildDropPath = CurrentDirectory,
+            PackageSupplier = PackageSuplier,
+            PackageName = PackageName,
+            PackageVersion = PackageVersion,
+            NamespaceBaseUri = NamespaceBaseUri,
+            BuildEngine = this.buildEngine.Object
+        };
+
+        // Act
+        var result = task.Execute();
+
+        // Assert
+        Assert.IsTrue(result);
+        Assert.IsTrue(File.Exists(manifestPath));
+
+        // Read and parse the manifest
+        var manifestContent = File.ReadAllText(manifestPath);
```

Comment:
It might be good to have a helper method for the chunk of code between lines 102-125, as it seems to be repeated in multiple test methods

Reply:
Added at d8d6951d9f58f133979ae09d7ffd039a560ecca9